sdk/src/tx/baseTxSender.ts

```diff
@@ -366,4 +367,33 @@ export abstract class BaseTxSender implements TxSender {
 	public getTimeoutCount(): number {
 		return this.timeoutCount;
 	}
+
+	public async checkConfirmationResultForError(txSig: string, result: RpcResponseAndContext<SignatureResult>) {
+		if (result.value.err) {
+			await this.reportTransactionError(txSig);
+		}
+	
+		return;
+	}
+
+	public async reportTransactionError(txSig: string) {
+		const transactionResult = await this.connection.getTransaction(txSig, {
+			maxSupportedTransactionVersion: 0
+		});
+
+		if (!transactionResult?.meta?.err) {
+			return undefined;
+		}
+		
+		const logs = transactionResult.meta.logMessages;
+
+		const lastLog = logs[logs.length - 1];
+
+		const friendlyMessage = lastLog?.match(/(failed:) (.+)/)?.[2];
+
+		throw new SendTransactionError(
```

Comment:
This would change the `TxSender` interface so that the send functions throw when they previously did not, which can catch users off guard.

suggestions:
* document the throwing behavior on the interface
* add an optional callback instead of throwing